pkg/generate/code/set_resource.go

```diff
@@ -110,23 +110,14 @@ func SetResource(
 	if op == nil {
 		return ""
 	}
-	outputShape := op.OutputRef.Shape
+	outputShape, _ := r.GetOutputShape(op)
```

Comment:
You'll probably want to handle this error, as it's critical. A panic might be a fair response if this returns an error - you cannot find the shape to use.

Reply:
`GetOutputShape` has a panic for that scenario. I kept logic consistent with previous behavior when refactoring

Reply:
But the method does throw other errors apart from that panic.
I think it makes sense to catch error and panic
(OR ugly version would be to panic for all errors inside `GetOutputShape` and remove `error` from return type)

Reply:
Fair enough... 
Since in the existing code outputshape is being returned as nil and then it is being handled in line 114, we do not need the error in method signature.

I would be onboard with removing error from method signature and keep the panic statement in method body